modules/social_features/social_group/modules/social_group_welcome_message/social_group_welcome_message.module

```diff
@@ -0,0 +1,150 @@
+<?php
+
+/**
+ * @file
+ * Primary module hooks for social_group_welcome_message module.
+ */
+
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Field\BaseFieldDefinition;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\group\Entity\GroupContentInterface;
+
+/**
+ * Implements hook_entity_base_field_info().
+ */
+function social_group_welcome_message_entity_base_field_info(EntityTypeInterface $entity_type) {
```

Comment:
Hey @OH17L ,

Is there a reason we're adding this as a base field here? Doesn't it make more sense to add this as a Field API field instead? That'll make it easier to modify per bundle if this is ever needed (e.g. for other group like types such as courses).

Reply:
Yeah, the reason to add this as a base field is that we're implementing welcome message to all group types and here (in hook_base_field_info) I'm able to do this with just one condition:
`if ($entity_type->id() === 'group')` (line 20 here)

Reply:
I discussed this with Ronald and we agree to keep it like this.

This is not because it's technically easier than adding fields to multiple bundles. The thought behind doing it this way is that the welcome message is something that fits the model of the group module.

This means that if something builds on the group entity that is not supposed to have this because it's not a group (for example a course) then that functionality should be implemented differently.

👍